extensions/amp-analytics/0.1/platform-variables.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {getService} from '../../../src/service';
+import {documentInfoFor} from '../../../src/document-info';
+import {parseUrl, removeFragment} from '../../../src/url';
+
+export function platformVariablesFor(window) {
+  return getService(window, 'amp-analytics-platform-variables', () => {
+    return platformVariables(window);
+  });
+}
+
+export function platformVariablesExpressionFor(window) {
+  const vars = platformVariablesFor(window);
+  return new RegExp('\{(' + Object.keys(vars).join('|') + ')\}', 'g');
+}
+
+function platformVariables(win) {
+  return {
+    /**
+     * Returns a random value for cache busters.
+     */
+    'random': () => {
```

Comment:
Right now they are pretty much the same but I intend to add a lot more stuff here that is not in URL replacement class. If all the variables defined in the spec are okay for url-replacement class, let me know and I can switch over to that. Another  issue is with the templating format. We would prefer the templates to be a little more than just uppercase words. I added {} for that reason. I can change it to {{}} or something else. If url-replacement can be configured to support {} or {{}} then let me know and I'll remove this.